src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -193,29 +198,26 @@ public virtual void Process (LinkContext context)
 
 		void Initialize ()
 		{
+			// Pre-load corelib and process its attribute XML first. This is necessary because the
+			// corelib attribute XML can contain modifications to other assemblies.
+			// We could just mark it here, but the attribute processing isn't necessarily tied to marking,
+			// so this would rely on implementation details of corelib.
+			var coreLib = _context.Resolve (PlatformAssemblies.CoreLib);
+			if (coreLib != null)
+				_context.CustomAttributes.EnsureProcessedAttributeXml (coreLib);
+
 			foreach (AssemblyDefinition assembly in _context.GetAssemblies ())
 				InitializeAssembly (assembly);
 		}
 
 		protected virtual void InitializeAssembly (AssemblyDefinition assembly)
 		{
 			var action = _context.Annotations.GetAction (assembly);
-			switch (action) {
-			case AssemblyAction.Copy:
-			case AssemblyAction.Save:
-				Tracer.AddDirectDependency (assembly, new DependencyInfo (DependencyKind.AssemblyAction, action), marked: false);
-				MarkEntireAssembly (assembly);
-				break;
-			case AssemblyAction.Link:
-			case AssemblyAction.AddBypassNGen:
-			case AssemblyAction.AddBypassNGenUsed:
-				MarkAssembly (assembly);
-
-				foreach (TypeDefinition type in assembly.MainModule.Types)
-					InitializeType (type);
+			if (action == AssemblyAction.Skip || action == AssemblyAction.Delete)
```

Comment:
.Delete check is redundant here

Reply:
How come? I don't think we want to initialize Delete assemblies.

Reply:
Can be removed in anticipation of Marek's change